miso-web/src/main/webapp/scripts/pane.js

```diff
@@ -0,0 +1,181 @@
+/*
+ * Copyright (c) 2012. The Genome Analysis Centre, Norwich, UK
+ * MISO project contacts: Robert Davey @ TGAC
+ * *********************************************************************
+ *
+ * This file is part of MISO.
+ *
+ * MISO is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * MISO is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with MISO.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * *********************************************************************
+ */
+
+PaneTarget = {};
+
+Pane = (function() {
+  var removeChildren = function(div) {
+    while (div.hasChildNodes()) {
+      div.removeChild(div.lastChild);
+    }
+  };
+
+  var initPaneDiv = function(div) {
+    div.setAttribute("class", "dashboard_widget");
+    removeChildren(div);
+
+    return div;
+  };
+
+  var createTitleDiv = function(title) {
+    var titleDiv = document.createElement("DIV");
+
+    titleDiv.setAttribute("class", "widget_title ui-corner-top");
+    titleDiv.innerText = title;
+
+    return titleDiv;
+  };
+
+  var createInputDiv = function(id) {
+    var input = document.createElement("INPUT");
+
+    input.setAttribute("type", "text");
+    input.setAttribute("size", "20");
+    input.setAttribute("id", id);
+
+    return input;
+  };
+
+  var createTitleDivWithSearch = function(inputId, title) {
+    var titleDiv = createTitleDiv(title);
+
+    titleDiv.append(createInputDiv(inputId));
+
+    return titleDiv;
+  };
+
+  var createContentDiv = function(id) {
+    var content = document.createElement("DIV");
+
+    content.setAttribute("class", "widget ui-corner-bottom");
+    content.setAttribute("id", id);
```

Comment:
If you keep references to the DIVs themselves, you don't need to track all these ids.

Reply:
:+1: 
This has been bugging me too. I'll try to fix this.